service/history/api/create_workflow_util.go

```diff
@@ -86,6 +91,23 @@ func NewWorkflowWithSignal(
 		return nil, err
 	}
 
+	if startRequest.StartRequest.WorkflowType.Name == scheduler.WorkflowType && shard.GetConfig().UseExperimentalHsmScheduler(startRequest.NamespaceId) {
+		// TODO(Tianyu): Should actually ensure the data converter is correct
```

Comment:
Switched to using the same data converter used on the client side: https://github.com/temporalio/temporal/blob/main/service/frontend/workflow_handler.go#L2932

In general though is there a better way to do this? This can easily go wrong if the data converter is changed on the other side.